crates/light-client-prover/src/da_block_handler.rs

```diff
@@ -149,10 +149,31 @@ where
 
         // Do any kind of ordering etc. on batch proofs here
         // If you do so, don't forget to do the same inside zk
+        let batch_prover_method_id = self
+            ._batch_proof_code_commitments_by_spec
+            .get(&SpecId::Genesis)
+            .expect("Batch proof code commitment not found");
+
+        let mut assumption = None;
+        let mut journal = None;
+        if !batch_proofs.is_empty() {
+            (assumption, journal) = match batch_proofs[0].clone() {
+                DaDataLightClient::Complete(proof) => {
+                    let data = proof.extract_proof();
```

Comment:
i think real and fake proof should be handled explicitly

*BUT* since we are now using fake receipt from risc0 and it requires dev mode to be enabled to verify those receipts, we don't actually need to seperate between real and fake proofs right? if so please create issue to remove the `accept_public_input` arg from the full nodes and update ci with the necessary env var.

Reply:
let's talk about this a bit on slack

Reply:
decided to merge #1392